File: app/src/main/java/com/example/bloodbank/DonationCentersAdapter.java
package com.example.bloodbank;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.List;

public class DonationCentersAdapter extends RecyclerView.Adapter<DonationCentersAdapter.CenterViewHolder> {
    private final List<DonationCenter> centersList;

    public DonationCentersAdapter(List<DonationCenter> centersList) {
        this.centersList = centersList;
    }

    @NonNull
    @Override
    public CenterViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_donation_center, parent, false);
        return new CenterViewHolder(view);
    }

    @Override
    public void onBindViewHolder(@NonNull CenterViewHolder holder, int position) {
        DonationCenter center = centersList.get(position);
        holder.nameTextView.setText(center.getName());
        holder.addressTextView.setText(center.getAddress());
        holder.phoneTextView.setText(center.getPhone());
        holder.emailTextView.setText(center.getEmail());
        holder.cityTextView.setText(center.getCity());
    }

    @Override
    public int getItemCount() {
        return centersList.size();
    }

    static class CenterViewHolder extends RecyclerView.ViewHolder {
        TextView nameTextView;
        TextView addressTextView;
        TextView phoneTextView;
        TextView emailTextView;
        TextView cityTextView;

        CenterViewHolder(View itemView) {
            super(itemView);
            nameTextView = itemView.findViewById(R.id.textViewName);
            addressTextView = itemView.findViewById(R.id.textViewAddress);
            phoneTextView = itemView.findViewById(R.id.textViewPhone);
            emailTextView = itemView.findViewById(R.id.textViewEmail);
            cityTextView = itemView.findViewById(R.id.textViewCity);
        }
    }
} 

File: app/src/main/java/com/example/bloodbank/adapter/DonationCentersAdapter.kt
package com.example.bloodbank.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.example.bloodbank.Model.DonationCenter
import com.example.bloodbank.R // Assuming R.layout.item_donation_center and TextView IDs are here

class DonationCentersAdapter(private val centersList: MutableList<DonationCenter>) :
    RecyclerView.Adapter<DonationCentersAdapter.CenterViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CenterViewHolder {
        val view = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_donation_center, parent, false)
        return CenterViewHolder(view)
    }

    override fun onBindViewHolder(holder: CenterViewHolder, position: Int) {
        val center = centersList[position]
        holder.bind(center)
    }

    override fun getItemCount(): Int = centersList.size

    inner class CenterViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        private val nameTextView: TextView = itemView.findViewById(R.id.textViewName)
        private val addressTextView: TextView = itemView.findViewById(R.id.textViewAddress)
        private val phoneTextView: TextView = itemView.findViewById(R.id.textViewPhone)
        private val emailTextView: TextView = itemView.findViewById(R.id.textViewEmail)
        private val cityTextView: TextView = itemView.findViewById(R.id.textViewCity)

        fun bind(center: DonationCenter) {
            nameTextView.text = center.name
            addressTextView.text = center.address
            phoneTextView.text = center.phone
            emailTextView.text = center.email
            cityTextView.text = center.city // Assuming 'city' is now a property of DonationCenter.kt
        }
    }
}